src/clarity-angular/datagrid/datagrid-column.ts

```diff
@@ -87,36 +100,92 @@ export class DatagridColumn extends DatagridFilterRegistrar<DatagridStringFilter
         } else if (value && !this.sorted) {
             this.sort();
         }
-    };
+    }
 
+    /**
+     * @deprecated This will be removed soon, in favor of the sortOrder mechanism
+     */
     @Output("clrDgSortedChange") public sortedChange = new EventEmitter<boolean>();
 
+    // deprecated: to be removed - END
+
+    /**
+     * Indicates how the column is currently sorted
+     */
+    private _sortOrder: SortOrder = SortOrder.Unsorted;
+    public get sortOrder() {
+        return this._sortOrder;
+    }
+
+    @Input("clrDgSortOrder")
+    public set sortOrder(value: SortOrder) {
+        // only if the incoming order is different from the current one
+        if (this._sortOrder === value) {
+            return;
+        }
+
+        switch (value) {
+            // the Unsorted case happens when the current state is either Asc or Desc
+            default: case SortOrder.Unsorted:
+                this._sort.clear();
+                break;
+            case SortOrder.Asc:
+                this.sort(false);
+                break;
+            case SortOrder.Desc:
+                this.sort(true);
+                break;
+        }
+    };
+
+    @Output("clrDgSortOrderChange") public sortOrderChange = new EventEmitter<SortOrder>();
+
     /**
      * Sorts the datagrid based on this column
      */
-    public sort() {
+    public sort(reverse?: boolean) {
         if (!this.sortable) {
             return;
         }
+
+        this._sort.toggle(this.sortBy, reverse);
+
+        // setting the private variable to not retrigger the setter logic
+        this._sortOrder = this._sort.reverse ? SortOrder.Desc : SortOrder.Asc;
+        this.sortOrderChange.emit(this._sortOrder);
```

Comment:
As I mentioned in response to your previous comment, we have 2 emission which correspond to 2 sources of change: this one handles the case where the column itself is the source of change (becomes sorted, or toggles asc/desc), whereas the one in the listener handles the case where another column becomes sorted, in which case this one becomes unsorted.

The fact that the change comes from such different sources means trying to handle them both in the same place leads to lifecycle errors. I'm fine having these 2 separate emissions, as long as it's just 2 of them.